packages/router/utils.ts

```diff
@@ -912,38 +918,38 @@ export class DeferredData {
   private trackPromise(
     key: string | number,
     value: Promise<unknown> | unknown
-  ) {
-    if (value instanceof Promise) {
-      this.pendingKeys.add(key);
-      value.then(
-        (data) => this.onSettle(key, null, data as unknown),
-        (error) => this.onSettle(key, error as unknown)
-      );
+  ): DeferredPromise | unknown {
+    if (!(value instanceof Promise)) {
+      return value;
     }
+
+    this.pendingKeys.add(key);
+
+    // We store a little wrapper promise that will be extended with
+    // _data/_error props upon resolve/reject
+    let deferredPromise: DeferredPromise = value.then(
+      (data) => this.onSettle(deferredPromise, key, null, data as unknown),
+      (error) => this.onSettle(deferredPromise, key, error as unknown)
+    );
+    Object.defineProperty(deferredPromise, "_tracked", { get: () => true });
+    return deferredPromise;
```

Comment:
`_tracked` is just so we don't register duplicate promise resolve/reject handlers

Reply:
Using `defineProperty` so these are non-enumerable etc.